### PR TITLE
Added metric exposing hash of all the confgs. This can be used to track config versions and updates.

### LINKED ISSUE #72: Add config version monitoring
Add a Prometheus metric exposing a hash of the config. 

Monitoring this metric will allow us to see configs running across the fleet.

### PR DESCRIPTION
**Pull request by grzkv**
## What issue is this change attempting to solve?
Fixes #72 

## How does this change solve the problem? Why is this the best approach?
MD5 hash on all configs. It's the shortest and unique enough.

## How can we be sure this works as expected?
Manual tests.

Note: This branch is on top of #89

### CODE REVIEW

File: cmd/nanotube/listen.go [hunk 1]
```diff
@@ -97,32 +88,37 @@ loop:
...
 		case conn := <-connCh:
 			wg.Add(1)
 			go readFromConnectionTCP(&wg, conn, queue, term, cfg, ms, lg)
 		}
 	}
-	lg.Info("Termination: stopped accepting new TCP connections. Starting to close incoming connections...")
+	lg.Info("Termination: Stopped accepting new TCP connections. Starting to close incoming connections...")
 	wg.Wait()
-	lg.Info("Termination: Finished incoming TCP connections...")
+	lg.Info("Termination: Finished previously accpted TCP connections.")
```
Comment (grlvrl) [inline cmd/nanotube/listen.go:105]: Typo in "accpted"

Comment (grzkv) [reply cmd/nanotube/listen.go:105]: Will fix later because this is actually in #79

File: cmd/nanotube/main.go [hunk 2]
```diff
@@ -75,21 +95,41 @@ func main() {
...
+			} else {
+				lg.Info("Reload: Config OK. Starting a new instance.")
+			}
+			pid, err := n.StartProcess()
+			if err != nil {
+				lg.Error("Reload: Failed to start new process", zap.Error(err))
+			} else {
+				lg.Info("Reload: Started new process. Moved FDs.", zap.Int("pid", pid))
+			}
+			lg.Info("Termination: Staring termination sequence")
```
Comment (grlvrl) [inline cmd/nanotube/main.go:120]: Typo in "Staring" - same in line 123

Comment (grzkv) [reply cmd/nanotube/main.go:120]: Thanks. I will upgrade my spell checker in VSCode) 
Same as the other one, it's from the other PR.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -123,16 +124,21 @@ func New(conf *conf.Main) *Prom {
...
+			Name:      "regex_duration_seconds",
+			Help:      "Time to evaluate each regex from configuration",
+		}, []string{"regex", "rule_type"}),
 		Version: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: "nanotube",
 			Name:      "version",
 			Help:      "Version info in label. Value should be always 1.",
 		}, []string{"version"}),
-		RegexDuration: prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		ConfVersion: prometheus.NewCounterVec(prometheus.CounterOpts{
```
Comment (avereha) [inline pkg/metrics/metrics.go:137]: Random thought: if we make it a gauge, we can store the hash as the value. 
Not sure if using the gauge would be easier, harder or the same...

Comment (grzkv) [reply pkg/metrics/metrics.go:137]: Thx for the idea.

**Pull request merged**